main/guides/zoe/contract-details.md

```diff
@@ -0,0 +1,121 @@
+# Durable Contract Details
+
+Zoe allows you to write smart contracts that manage interactions between cooperative, but possibly untrusting,
+parties. Some contracts perform a single simple task like trading one good for another, and then are
+done. Others stay around and manage assets and interactions over time. The first kind don't need persistence
+or the ability to upgrade, while the second kind need to make use of more sophisticated functionality.
+
+When a contract is intended to continue running and serving many customers over a long time, its objects and
+data need to be persistent, its owners or managers may need to be able to adjust parameters or perform other
+governance actions, and it needs to be upgradeable so that the code can adapt over time.
```

Comment:
"needs"? as in _every_ long-running multi-customer contract _must_ be upgradable?

Immutable contracts are ideal in some ways, no?